### PR TITLE
Remove "Demo" from example titles (part 2)

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

More cleanup like #18858.

### CODE REVIEW

File: examples/event_handling/figure_axes_enter_leave.py [hunk 1]
```diff
@@ -32,24 +32,13 @@ def on_leave_figure(event):
...
+fig.suptitle('mouse hover over figure or axes to trigger events')
 
-fig1.canvas.mpl_connect('figure_enter_event', on_enter_figure)
-fig1.canvas.mpl_connect('figure_leave_event', on_leave_figure)
-fig1.canvas.mpl_connect('axes_enter_event', on_enter_axes)
-fig1.canvas.mpl_connect('axes_leave_event', on_leave_axes)
-
-###############################################################################
-
-fig2, (ax, ax2) = plt.subplots(2, 1)
```
Comment (timhoffm) [inline examples/event_handling/figure_axes_enter_leave.py:47]: fig2 is just an exact copy of the fig1 example.

**Review by story645 — CHANGES_REQUESTED** [review]
It looks like a stray notebook (untitled6) got included?

File: examples/event_handling/coords_demo.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 """
-===========
-Coords demo
-===========
+==========================
+Mouse move and click event
+==========================
```
Comment (QuLogic) [inline examples/event_handling/coords_demo.py:4]:
```suggestion
===========================
Mouse move and click events
===========================
```

File: examples/event_handling/figure_axes_enter_leave.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
 """
-=======================
-Figure Axes Enter Leave
-=======================
+=================================
+Figure/Axes enter and leave event
+=================================
```
Comment (QuLogic) [inline examples/event_handling/figure_axes_enter_leave.py:4]:
```suggestion
==================================
Figure/Axes enter and leave events
==================================
```

File: examples/subplots_axes_and_figures/multiple_figs_demo.py [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
+`matplotlib.pyplot` uses the concept of a *current figure* and *current axes*.
+Figures are identified via a figure number that is passed to `~.pyplot.figure`.
+The figure with the given number is set as *current figure*. Additionally, if
+no figure with the number exists, a new one is created.
+
+.. note::
+
+    We discourage working with multiple figures in pyplot because managing
+    the *current figure* is cumbersome and error-prone. Instead, we recommend
+    to use the object-oriented approach and call methods on figure and Axes
```
Comment (QuLogic) [inline examples/subplots_axes_and_figures/multiple_figs_demo.py:15]: This is definitely talking about the class.
```suggestion
    to use the object-oriented approach and call methods on Figure and Axes
```

**Review by story645 — APPROVED** [review]

**Pull request merged**